src/main/scala/nl/biopet/tools/bamstats/generate/Generate.scala

```diff
@@ -89,46 +89,48 @@ object Generate extends ToolCommand[Args] {
   def extractStats(bamFile: File,
                    bedFile: Option[File],
                    sequenceDict: SAMSequenceDictionary,
-                   excludePartialReads: Boolean = false,
-                   getUnmappedReads: Boolean = false): GroupStats = {
+                   scatterMode: Boolean = false,
+                   onlyUnmapped: Boolean = false): GroupStats = {
     val samReader: SamReader =
       SamReaderFactory.makeDefault().open(bamFile)
     val stats = GroupStats()
 
-    // Read stats from regions
-    bedFile.foreach { bed =>
-      // Combine overlap because we do not want the same reads to be counted twice
-      val regions =
-        BedRecordList.fromFile(bed).combineOverlap.validateContigs(sequenceDict)
-      regions.allRecords.foreach { bedRecord =>
-        val samRecordIterator: SAMRecordIterator =
-          // Is contained = excludePartialReads an option here?
-          // I think not, because this also excludes reads that originate within the region, but end
-          // in another region.
-          samReader.query(bedRecord.chr, bedRecord.start, bedRecord.end, false)
-        samRecordIterator.foreach { samRecord =>
-          // Read based stats
-          // If excludePartialReads is false, continue.
-          // If excludePartialReads is true, determine whether the alignment start is within the region.
-          if (!excludePartialReads || samRecord.getAlignmentStart > bedRecord.start && samRecord.getAlignmentStart <= bedRecord.end) {
-            stats.loadRecord(samRecord)
-          }
-        }
-      }
-    }
+    // The below logic makes sure that either the unmapped reads, reads from a region, or all reads are returned.
+    // Never a combination of the above.
 
-    // Read stats from unmappedReads
-    if (getUnmappedReads) {
+    // Read stats from unmappedReads.
+    if (onlyUnmapped) {
```

Comment:
I think this can be done in 1 match statement with an exception if unmapped and a bed file is set:
```scala
bedFile match {
  case Some(_) if onlyUnmapped => throw new IlegalArgumentException("Text")
  case None if onlyUnmapped => samReader.queryUnmapped().foreach(stats.loadRecord)
  case Some(file) => // Code with bed file
  case _ => samReader.iterator().foreach(stats.loadRecord)
}
```